ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -239,4 +239,100 @@ public static synchronized String getSchemaDefinitionString(Schema schema) {
     return stringBuilder.toString();
   }
 
+  public static String getSQLTypeName(Schema schema) {
+    switch (schema.type()) {
+      case INT32:
+        return "INT";
+      case INT64:
+        return "BIGINT";
+      case FLOAT32:
+      case FLOAT64:
+        return "DOUBLE";
+      case BOOLEAN:
+        return "BOOLEAN";
+      case STRING:
+        return "VARCHAR";
+      case ARRAY:
+        return "ARRAY<" + getSQLTypeName(schema.valueSchema()) + ">";
+      case MAP:
+        return "MAP<" + getSQLTypeName(schema.keySchema()) + "," + getSQLTypeName(schema
+                                                                                       .valueSchema()) + ">";
+      default:
+        throw new KsqlException(String.format("Invalid type in schema: %s.", schema.toString()));
+    }
+  }
+
+  public static  synchronized String buildAvroSchema(final Schema schema, String name) {
+    StringBuilder stringBuilder = new StringBuilder("{\n\t\"namespace\": \"ksql\",\n");
+    stringBuilder.append("\t\"name\": \"" + name + "\",\n");
+    stringBuilder.append("\t\"type\": \"record\",\n");
+    stringBuilder.append("\t\"fields\": [\n");
+    boolean addCamma = false;
+    Set<String> fieldNameSet = new HashSet<>();
+    for (Field field : schema.fields()) {
+      if (addCamma) {
+        stringBuilder.append(",\n");
+      } else {
+        addCamma = true;
+      }
+      String fieldName = field.name().replace(".", "_");
+      while (fieldNameSet.contains(fieldName)) {
+        fieldName = fieldName + "_";
+      }
+      fieldNameSet.add(fieldName);
+      stringBuilder
+          .append("\t\t{\"name\": \"" + fieldName + "\", \"type\": "
+                  + getAvroTypeName(field.schema()) + "}");
+    }
+    stringBuilder.append("\n\t]\n");
+    stringBuilder.append("}");
+    return stringBuilder.toString();
+  }
+
+  private static String getAvroTypeName(final Schema schema) {
+    switch (schema.type()) {
+      case STRING:
+        return "\"string\"";
+      case BOOLEAN:
+        return "\"boolean\"";
+      case INT32:
+        return "\"int\"";
+      case INT64:
+        return "\"long\"";
+      case FLOAT64:
+        return "\"double\"";
+      default:
+        if (schema.type() == Schema.Type.ARRAY) {
+          return "{\"type\": \"array\", \"items\": "
+                 + getAvroTypeName(schema.valueSchema()) + "}";
+        } else if (schema.type() == Schema.Type.MAP) {
+          return "{\"type\": \"map\", \"values\": "
+                 + getAvroTypeName(schema.valueSchema()) + "}";
+        }
+        throw new KsqlException("Unsupported AVRO type: " + schema.type().name());
+    }
+  }
+
+  public static synchronized Schema getAvroSerdeKsqlSchema(Schema schema) {
```

Comment:
This method name and the next one isn't very helpful. How does the input relate to the output. By the looks of it, only fields are renamed. Why do we need to do this? Some comments or java doc would help.

Reply:
Added comments to the code.